addons/udes_stock/tests/test_stock_move_line.py

```diff
@@ -540,6 +540,511 @@ def test_over_mark_error(self):
+        )
+        for mls, mls_values in res.items():
+            mls.mark_as_done(mls_values)
+        self.assertIn(picking.move_line_ids.lot_id.mapped("name")[0], ["1", "5", "4"])
+        self.assertIn(picking.move_line_ids.lot_id.mapped("name")[1], ["1", "5", "4"])
+        self.assertIn(picking.move_line_ids.lot_id.mapped("name")[2], ["1", "5", "4"])
+        self.assertEqual(picking.move_line_ids.mapped("product_uom_qty"), [1.0, 1.0, 1.0])
+        self.assertEqual(picking.move_line_ids.mapped("qty_done"), [1.0, 1.0, 1.0])
+
+    def test_swap_lot_lot_name_on_single_move_line(self):
+        """
+        When prepare is given a differrent lot name for a single lot product move_line, and u_tracked_product_swap is turned on, then the resultant move_line value will
+        have swapped lot_names
+        """
+        self.picking_type_pick.u_allow_swapping_tracked_products = True
+        for lot_num in range(1, 6):
+            self.create_quant(
+                self.tangerine.id,
+                location_id=self.test_stock_location_01.id,
+                qty=10,
+                lot_name=f"{lot_num}",
+            )
+        product_info = [{"product": self.tangerine, "uom_qty": 10}]
+        picking = self.create_picking(
+            self.picking_type_pick, products_info=product_info, assign=True
+        )
+
+        product_ids = [{"barcode": "productTangerine", "uom_qty": 10, "lot_names": ["2"]}]
+        location = self.test_stock_location_01
+        pallet_to_pick_onto = self.create_package(name="UDES1")
+
+        res = picking.move_line_ids.prepare(
+            product_ids=product_ids, location=location, result_package=pallet_to_pick_onto
+        )
+        self.assertEqual(res[picking.move_line_ids]["lot_name"], "2")
+        self.assertEqual(res[picking.move_line_ids]["product_uom_qty"], 10)
+        self.assertEqual(res[picking.move_line_ids]["qty_done"], 10)
+        for mls, mls_values in res.items():
+            mls.mark_as_done(mls_values)
+        self.assertEqual(picking.move_line_ids.lot_id.mapped("name"), ["2"])
+
+    def test_swapping_the_lots_on_two_move_lines(self):
+        """
+        Swap lot when new lot is actually reserved on the move_line of another picking
+        """
+        self.picking_type_pick.u_allow_swapping_tracked_products = True
+        for lot_num in range(1, 6):
+            self.create_quant(
+                self.tangerine.id,
+                location_id=self.test_stock_location_01.id,
+                qty=10,
+                lot_name=f"{lot_num}",
+            )
+        product_info = [{"product": self.tangerine, "uom_qty": 10}]
+        first_picking = self.create_picking(
+            self.picking_type_pick, products_info=product_info, assign=True
+        )
+        second_picking = self.create_picking(
+            self.picking_type_pick, products_info=product_info, assign=True
+        )
+
```

Comment:
Would be good to check that lot of the first picking move line is "1" and of the second picking is "2".